skbio/metadata/tests/test_intersection.py

```diff
@@ -189,6 +189,19 @@ def test_find(self):
         r = self.iv.find(100, 200)
         self.assertEqual(len(r), 5)
 
+    def test_edge_cases(self):
+        iv = IntervalTree()
+        iv.insert(1, 1, 'foo')
+        iv.insert(3, 7, 'spam')
+        self.assertEqual(iv.find(0, 1), [])
+        self.assertEqual(iv.find(1, 1), ['foo'])
+        self.assertEqual(iv.find(1, 2), ['foo'])
+        self.assertEqual(iv.find(2, 3), [])
+        self.assertEqual(iv.find(3, 3), ['spam'])
+        self.assertEqual(iv.find(3, 4), ['spam'])
+        self.assertEqual(iv.find(6, 7), ['spam'])
+        self.assertEqual(iv.find(7, 7), [])
```

Comment:
Add a test that `iv.find(0, 8)` finds `['foo', 'spam']` just to make sure that "point ranges" interact as expected with normal ranges.